uftrace.h

```diff
@@ -422,6 +422,8 @@ enum uftrace_msg_type {
 /* Options sent by the client to the agent */
 enum agent_opt {
 	AGENT_CLOSE = (1U << 0), /* Close the connection with the client */
+
+	AGENT_TRACE = (1U << 5), /* Tracing state (on/off) */
```

Comment:
Do you intend to have a complex option with bitwise-OR?  Otherwise we can just use natural numbers.

Reply:
Yes I want to introduce a capability system, so the client can request supported features from the agent. For example changing filters could require locks, and thus be disabled on certain libmcount builds.

Reply:
I think the client should receive a response from the agent after sending the message to check if it's supported.  But I'm not sure if it needs OR-ing the messages like `UAM_CLOSE | UAM_TRACE` or something.

Reply:
Ok, the way I intended it was to ask for all the supported features, and then only forwarding options that are supported. What do you think?
You can check the relevant commit on #1645: b6710b2.

Reply:
Yeah, I mean it too.  Then we can define the enum simply like:

```c
enum uftrace_agent_message {
    UAM_CLOSE = 0,
    UAM_TRACE = 1,
    ...
};
```

Reply:
Also, I think agent need to send 'ACK' back to client after receiving a message so that it can confirm whether it's handled successfully.

Reply:
I'm OK with having simple integers in the enum. I'm going to send `ACK` from the agent for every message from the client.

> I think the client should receive a response from the agent after sending the message to check if it's supported.

Something like a handshake message prior to sending other messages?

Reply:
Right, maybe we need to design the communication protocol first.  I don't want to make things complicated but I guess we will add more messages later.  Also not sure if multiple clients want to talk to the agent at the same time.

Reply:
I'm thinking about discussing this in a separate issue, is it ok?

Reply:
Sure!